src/ucp/core/ucp_worker.c

```diff
@@ -1168,6 +1168,60 @@ void ucp_worker_iface_cleanup(ucp_worker_iface_t *wiface)
     ucp_worker_uct_iface_close(wiface);
 }
 
+static ucs_status_t ucp_worker_add_resource_cms(ucp_worker_h worker)
+{
+    ucp_context_h   context  = worker->context;
+    ucs_status_t    status   = UCS_OK;
+    uct_cm_h        *cms_tmp;
+    ucp_rsc_index_t i;
+
+    if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    cms_tmp = ucs_alloca(context->num_cmpts);
```

Comment:
something wrong here: allocated ```ucs_alloca(context->num_cmpts)```, but copied ```memcpy(worker->cms, cms_tmp```, **worker->num_cms * sizeof(*worker->cms)**)
is it ok?

Reply:
yeah, missing sizeof